pl_bolts/models/rl/per_dqn_model.py

```diff
@@ -59,6 +59,52 @@ class PERDQN(DQN):
         .. note:: Currently only supports CPU and single GPU training with `distributed_backend=dp`
 
         """
+    def train_batch(self) -> Tuple[Tuple, torch.Tensor, torch.Tensor]:
+        """
+        Contains the logic for generating a new batch of data to be passed to the DataLoader
+
+        Returns:
+            yields a Experience tuple.
+        """
+        self.agent.update_epsilon(self.total_steps)
+
+        # take a step in the env
+        exp, reward, done = self.source.step(self.device)
+        self.episode_steps += 1
+        self.total_steps += 1
+
+        self.reward_sum += exp.reward
+        self.episode_reward += reward
+
+        # gather the experience data
+        self.buffer.append(exp)
+
+        if done:
+            # tracking metrics
+            self.episode_count += 1
+            self.reward_list.append(self.total_reward)
+            self.avg_reward = sum(self.reward_list[-self.avg_reward_len:]) / self.avg_reward_len
+            self.total_reward = self.episode_reward
+            self.total_episode_steps = self.episode_steps
+
+            self.logger.experiment.add_scalar("reward", self.total_reward, self.total_steps)
+
+            # reset metrics
+            self.episode_reward = 0
+            self.episode_steps = 0
```

Comment:
shall this be rater at the beginning rather than the end?

Reply:
Im not sure I understand

Reply:
that you reset the `episode_steps` and the other when it is done... so shall it be more logical to reset it before you start rather?

Reply:
Ah I see. The `done` is a local variable that is retrieved after taking a step on line 72, so the done check must come after that, so I think it makes more sense to do it at the end